indico/modules/events/papers/models/papers.py

```diff
@@ -66,3 +68,45 @@ def state(self):
     @state.setter
     def state(self, state):
         self.last_revision.state = state
+
+    def is_in_final_state(self):
+        return self.state in {PaperRevisionState.accepted or PaperRevisionState.rejected}
+
+    def can_comment(self, user):
+        if not user:
```

Comment:
Two lines in every single method that is used to do protection check, which are starting to sprawl and it's easier to forget adding the check than the decorator. We could make it less obscure and call it `@require_user`.